system/jlib/jthread.cpp

```diff
@@ -2085,11 +2140,10 @@ protected: friend class PipeWriterThread;
                 if (chdir(dir) == -1)
                     throw MakeStringException(-1, "CLinuxPipeProcess::run: could not change dir to %s", dir.get());
             }
-            ForEachItemIn(idx, envVars)
-            {
-                ::setenv(envVars.item(idx), envValues.item(idx), 1);
-            }
-            execvp(argv[0],argv);
+            if (envp.length())
+                execve(argv[0], argv, (char *const *) envp.detach());
```

Comment:
curious what happens if you didn't detach() ... ?
It would not call the StringAttr dtor in child, so would be safe anyway (?) , but it's clearer like this..